Mod Events | Encounters

Ink230 opened the pull request:
### SubmapEncounterRateEvent
Purpose
- Permit the encounter rate to be modified on each step

Example Uses
- Change the base encounter rate on each step to a random value
- Change the base encounter rate per submapId

### SubmapEncounterAccumulatorEvent
Purpose
- Modify the formula used to accumulate the encounter threshold
- Permit adjustment of the encounter threshold

Example Uses
- Implement one's own accumulator formula
- Interact with the existing accumulated value

### SubmapGenerateEncounterEvent
Purpose
- Modify the encounterId
- Modify the battleStageId

Example Uses
- Provide a random encounterId or different encounterId based on some logic
- Provide a random battleStageId or different battleStageId based on some logic
- Decide on the encounterId/BattleStageId depending on the battleStageId/encounterId, respectively

### BattleEnecounterStageDataEvent
Purpose
- Load a different StageData2c row
- Modify the received StageData2c row
- Create a new StageData2c

Example Uses
- Dynamically modify StageData properties on external logic
- Map existing StageData to a modification list
- Create and return one's own data set of StageData entries more easily
- Modify Escape % dynamically

### BattleMusicEvent
Purpose
- Modify the victoryType
- Modify the musicIndex
  
Example Uses
- Play a specific victoryType despite the stagedata, stageid, encounterId
- Play a specific musicIndex despite the stagedata, stageid, encounterId

### ~~UtilitySimpleRandEvent~~
~~Purpose~~
- ~~Modify SimpleRand()'s implementation for game logic~~
- ~~Override SimpleRands()'s implementation for game logic~~

~~Example Uses~~
- ~~Override SimpleRand to provide one's own implementation of RNG for game logic~~
- ~~Can be used to offer various randomization algorithms over SimpleRand~~

LordMonoxide reviewed (changes requested):
Update for latest main, remove random event, and it's good

Ink230 commented:
- Confirmed functionality after rebasing to latest main
- Removed UtilitySimpleRandEvent
- Updated stagedata event property names to their renamed versions from us confirming what they were for
